onnxmltools/convert/common/_topology.py

```diff
@@ -639,6 +640,22 @@ def convert_topology(topology, model_name, doc_string, targeted_onnx):
         if name in other_inputs:
             container.add_input(other_inputs[name])
 
+    # Add metadata properties and channel denotation to image tensors
+    metadata_props = {}
+    image_inputs = []
+    for variable in topology.find_root_and_sink_variables():
+        color_space = getattr(variable.type, 'color_space', None)
+        if color_space:
+            image_inputs.append(variable.raw_name)
+            pixel_format = color_space_to_pixel_format(color_space)
+            if metadata_props.get('Image.BitmapPixelFormat', pixel_format) != pixel_format:
+                print('Warning: conflicting pixel formats found. In ONNX, all input/output images must use the same pixel format.')
+            metadata_props = {
```

Comment:
these default value aligns with Core ML, or other standard?

Reply:
SRGB and 0-255 are the most commonly used formats for input images. They are also the values used in type denotation sample in the ONNX documentation. We believe it's a sane default if the user doesn't provide any information.

Reply:
Are those information (e.g., `Image.BitmapPixelFormat`) formal definitions in ONNX? If no, I'd prefer to leave this job (adding metadata strings) to the parser for source models (e.g., for Core ML models, we have [this](https://github.com/onnx/onnxmltools/blob/f8743958da9594488c29fdffb501f53f2f5ecc5d/onnxmltools/convert/coreml/_parse.py#L49))

Reply:
Also, this metadata dictionary gets overwritten if there are multiple input images..

Reply:
I see. Thanks. Please see also my comment for `add_metadata_props(onnx_model, metadata_props)` below. My feeling is that we should add such information at the earliest place, which is the parsing function for different frameworks (e.g., Core ML and scikit-learn) and maintain it throughout the compiling procedure. Although I am still open to your current approach because Core ML is the only model which needs medata.